Mathlib/GroupTheory/Perm/MaximalSubgroups.lean

```diff
@@ -0,0 +1,430 @@
+/-
+Copyright (c) 2025 Antoine Chambert-Loir. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir
+-/
+import Mathlib.GroupTheory.GroupAction.Jordan
+import Mathlib.GroupTheory.SpecificGroups.Cyclic
+import Mathlib.GroupTheory.Subgroup.Simple
+import Mathlib.GroupTheory.GroupAction.SubMulAction.OfFixingSubgroup
+
+/-! # Maximal subgroups of the symmetric groups
+
+* `Equiv.Perm.isCoatom_stabilizer`: when `s : set α` is
+  not empty, nor its complementary subset,
+  and if the cardinality of `s` is not half of that of `α`,
+  then `MulAction.stabilizer (Equiv.Perm α) s` is
+  a maximal subgroup of the symmetric group `Equiv.Perm α`.
+
+  This is the *intransitive case* of the O'Nan-Scott classification.
+
+## TODO
+
+  * Appplication to primitivity of the action
+    of `Equiv.Perm α` on finite combinations of `α`.
+
+  * Formalize the other cases of the classification.
+  The next one should be the *imprimitive case*.
+
+## Reference
+
+The argument is taken from [M. Liebeck, C. Praeger, J. Saxl,
+*A classification of the maximal subgroups of the finite
+alternating and symmetric groups*, 1987][LiebeckPraegerSaxl-1987].
+-/
+
+open scoped Pointwise
+
+open Set
+
+theorem Subtype.image_preimage_of_val {α : Type*} {s B : Set α} (h : B ⊆ s) :
```

Comment:
```suggestion
theorem Subtype.val_image_preimage_cancel_of_le {α : Type*} {s B : Set α} (h : B ⊆ s) :
```
maybe?  I am not a naming expert, but at least the `Subtype.val` part looks reasonable!